CenterForOpenScience/osf.io
# [ENG-3856] working addon-usage report

## Purpose
get some insight into how many users and nodes are set up to use various storage addons,
without the report taking days to run.


## Changes
- rewrite the addon-usage report to use fewer queries (`O(addon-count)` instead of `O(node-count)`)
  - rename internally to `storage_addon_usage`, since there are other kinds of addons
- update `daily_reporters_go` management command
  - add `--filter` param to selectively run reporters
  - fix `--date` param parsing
  - avoid error trying to print errors
- when running the `daily_reporters_go` management command thru the admin web ui, run it in a celery task instead of blocking 'til all reporters finish (the rewritten reporter may still take a few minutes)


## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify `api:_/metrics/reports/storage_addon_usage/recent/` has reasonable-looking data
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Documentation



## Side Effects



## Ticket
[ENG-3856]



[ENG-3856]: https://openscience.atlassian.net/browse/ENG-3856?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
